packages/mongo-nav/src/org-nav/OrgNav.tsx

```diff
@@ -243,75 +246,95 @@ export default function OrgNav({
         constructOrganizationURL={constructOrganizationURL}
         urls={urls}
         onChange={onOrganizationChange}
-        isActive={activeNav === NavElement.OrgSettings}
+        isActive={activeNav === NavElement.OrgNavOrgSettings}
         loading={!current}
         disabled={disabled}
         isOnPrem={onPremEnabled}
       />
 
-      {!disabled && (
+      {renderBadgeItem()}
+
+      {!disabled && !isMobile && (
         <>
-          {badgeItem}
-
-          {!isMobile && (
-            <>
-              <OrgNavLink
-                href={current && orgNav.accessManager}
-                isActive={activeNav === NavElement.AccessManager}
-                loading={!current}
-                data-testid="org-nav-access-manager"
-              >
-                Access Manager
-              </OrgNavLink>
-
-              <IconButton
-                ariaLabel="Dropdown"
-                active={accessManagerOpen}
-                disabled={!current}
-                onClick={() => setAccessManagerOpen(curr => !curr)}
-              >
-                <Icon glyph={accessManagerOpen ? 'CaretUp' : 'CaretDown'} />
-
-                {current && (
-                  <Menu
-                    open={accessManagerOpen}
-                    setOpen={setAccessManagerOpen}
-                    usePortal={false}
-                    className={accessManagerMenuContainer}
-                  >
-                    <p className={accessManagerMenuItem}>
-                      <strong>Organization Access:</strong> {current.orgName}
-                    </p>
-
-                    <p className={accessManagerMenuItem}>
-                      <strong>Project Access: </strong>
-                      {currentProjectName ?? 'None'}
-                    </p>
-                  </Menu>
-                )}
-              </IconButton>
-
-              <OrgNavLink
-                href={current && orgNav.support}
-                isActive={activeNav === NavElement.Support}
-                loading={!current}
-                className={supportContainer}
-                data-testid="org-nav-support"
+          <OrgNavLink
+            href={current && orgNav.accessManager}
+            isActive={activeNav === NavElement.OrgNavAccessManager}
+            loading={!current}
+            data-testid="org-nav-access-manager"
+            onClick={e => onElementClick(NavElement.OrgNavAccessManager, e)}
+          >
+            Access Manager
+          </OrgNavLink>
+
+          <IconButton
+            ariaLabel="Dropdown"
+            active={accessManagerOpen}
+            disabled={!current}
+            data-testid="org-nav-dropdown"
+            onClick={(e: React.MouseEvent) => {
+              onElementClick(NavElement.OrgNavDropdown, e);
```

Comment:
So here's an idea of a refactor for the value we get back from `useOnElementClick`.

What if we wrap the value given as a prop into a function that looks like this:

```jsx
const wrappedOnElementClick = (navElement: NavElement, additionalWork: (e) => void) {
   return (e) => {
        onElementClick(NavElement.OrgNavDropdown, e);
        return additionalWork(e);
  };
};

// So you could use it one of these two ways:
<MyComponent onClick={onElementClick(NavElement.OrgNavDropdown)} />
// or
<MyComponent onClick={onElementClick(NavElement.OrgNavDropdown, () => setStateThing())} />
```

Reply:
👍 woo!